doc/adr/0001-record-architecture-decisions.rst

```diff
@@ -0,0 +1,106 @@
+ADR 1: Record architectural decisions
+=====================================
+
+Date: 2022-06-08
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+We are in search for a means to document our architectural and design decisions
+for all of our components.
+In order to do that, there is practice called architectural decision records ("ADR"),
+that we can integrate into our workflow.
+
+This does not replace actual architecture documentation, but provides contributors:
+
+* the means to understand architectural and design decisions that were made
+* a framework for proposing changes to the current architecture
+
+For each decision, it is important to consider the following factors:
+
+* what we have decided to do
+* why we have made this decision
+* what we expect the impact of this decision to be
+* what we have learned in the process
+
+As we're already using `rST <https://docutils.sourceforge.io/rst.html>`_, `Sphinxdoc <https://www.sphinx-doc.org/en/master/>`_ and `readthedocs <https://readthedocs.org/>`_, it would be practical to
+integrate these ADRs as part of our current documentation infrastructure.
+
+Decision
+--------
+
+* We will use ADRs to document, propose and discuss
+  any important or significant architectural and design decisions.
+
+* The ADR format will follow the format described in `Consequences`_ section.
+
+* We will follow the convention of storing those ADRs as Markdown formatted
+  documents stored under the `docs/adr` directory, as exemplified in Nat Pryce's
+  `adr-tools <https://github.com/npryce/adr-tools>`_. This does not imply that we will
+  be using `adr-tools` itself, as we might diverge from the proposed structure.
+
+* We will keep rejected ADRs
+
+* TODO: Add decision about **when** we should create an ADR (see PR discussions)
+
+Consequences
+------------
+
+ADRs should be written using the template described in the `ADR template`_ which comes from
+Chapter 6.5.2 (*A Template for Documenting Architectural Decisions*) of
+*Documenting Software Architectures: Views and Beyond (2nd Edition)*.
+
+However, the mandatory sections are *Title*, *Issue*, *Decision*, *Status*, *Implications*.
+The rest are optional.
+
+Another good reference is the article `Architecture Decision Records <https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions>`_ by Michael Nygard (Nov. 15, 2011).
+
+ADR template
+^^^^^^^^^^^^
+
+What follows is the ADR format (adapted from the book).
+
+============================== ===========
```

Comment:
Tables in rST are really annoying to make...

Reply:
:people_hugging: